core/src/com/unciv/logic/city/CityReligion.kt

```diff
@@ -19,25 +37,69 @@ class CityInfoReligionManager: Counter<String>() {
     }
     
     fun getNumberOfFollowers(): Counter<String> {
-        val totalInfluence = values.sum()
+        val totalInfluence = pressures.values.sum()
         val population = cityInfo.population.population
         if (totalInfluence > 100 * population) {
             val toReturn = Counter<String>()
-            for ((key, value) in this)
+            for ((key, value) in pressures)
                 if (value > 100)
                     toReturn.add(key, value / 100)
             return toReturn
         }
 
         val toReturn = Counter<String>()
 
-        for ((key, value) in this) {
+        for ((key, value) in pressures) {
             val percentage = value.toFloat() / totalInfluence
             val relativePopulation = (percentage * population).roundToInt()
             toReturn.add(key, relativePopulation)
         }
         return toReturn
     }
+    
+    fun clearAllPressures() {
+        pressures.clear()
+    }
+    
+    fun addPressure(religionName: String, amount: Int) {
+        val currentMajorityReligion = getMajorityReligion()
+        pressures.add(religionName, amount)
+        val newMajorityReligion = getMajorityReligion()
+        if (currentMajorityReligion != newMajorityReligion && newMajorityReligion != null) {
+            triggerReligionAdoption(newMajorityReligion)
+        }
+    }
+
+    private fun triggerReligionAdoption(newMajorityReligion: String) {
+        cityInfo.civInfo.addNotification("Your city [${cityInfo.name}] was converted to [$newMajorityReligion]!", NotificationIcon.Faith)
+
+        if (newMajorityReligion in religionsAtSomePointAdopted) return
+        
+        
+        val religionOwningCiv = cityInfo.civInfo.gameInfo.getCivilization(cityInfo.civInfo.gameInfo.religions[newMajorityReligion]!!.foundingCivName)
+        
+        for (unique in cityInfo.civInfo.gameInfo.religions[newMajorityReligion]!!.getFounderUniques()) {
+            if (unique.placeholderText == "[] when a city adopts this religion for the first time (modified by game speed)") {
+                val statsGranted = unique.stats.times(cityInfo.civInfo.gameInfo.gameParameters.gameSpeed.modifier)
+                religionOwningCiv.addStats(statsGranted)
+                
+                if (cityInfo.location in religionOwningCiv.exploredTiles)
+                    religionOwningCiv.addNotification("You gained [$statsGranted] as your religion was spread to [${cityInfo.name}]", cityInfo.location, NotificationIcon.Faith)
+                else
+                    religionOwningCiv.addNotification("You gained [$statsGranted] as your religion was spread to an unknown city", NotificationIcon.Faith)
+            }
+            
+            if (unique.placeholderText == "[] when a city adopts this religion for the first time") {
```

Comment:
ah - implement both but only one used in json... OK, but code duplication could be folded away.